File: Selenium/Activities/Selenium-Java/activities/Activity2Xpath.java
/*
Using Selenium:

Open a new browser to https://training-support.net/selenium/login-form
Get the title of the page and print it to the console.
Find the username field using any locator and enter "admin" into it.
Find the password field using any locator and enter "password" into it.
Find the "Log in" button using any locator and click it.
Close the browser.

*/


package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity2Xpath {

    public static void main(String[] args) throws InterruptedException{
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE,"/dev/null");
        // Set up Firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net");
        //get the title
        System.out.println("Title of webpage is" + driver.getTitle());

        // click on Login button on top right side
        // Open the page
        driver.get("https://www.training-support.net/selenium/login-form");
        // Print the title of the page
        System.out.println("Home page title: " + driver.getTitle());
        //driver.findElement(By.className("ui button")).click();
        Thread.sleep(2000);


        // Find the username field and enter the username
        driver.findElement(By.xpath("/html/body/div[4]/div/div[2]/form/div[1]/input")).sendKeys("admin");
        // Find the password field and enter the password
        driver.findElement(By.xpath("/html/body/div[4]/div/div[2]/form/div[2]/input")).sendKeys("password");
        // Find the login button and click it
        driver.findElement(By.xpath("/html/body/div[4]/div/div[2]/form/button")).click();

        // Print the confirmation message
        String message = driver.findElement(By.id("action-confirmation")).getText();
        System.out.println("Login message: " + message);

        // Close the browser
        driver.close();


    }
}

File: Selenium/Activities/Selenium-Python/activities/Activity2_XPATH.py
""" Using Selenium:

Open a new browser to https://training-support.net/selenium/login-form
Get the title of the page and print it to the console.
Find the username field using XPATH locator and enter "admin" into it.
Find the password field using XPATH locator and enter "password" into it.
Find the "Log in" button using XPATH locator and click it.
Close the browser."""

# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.core import driver
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service1 = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service1) as driver:
    # Open the browser to the URL
    driver.get("https://www.training-support.net/selenium/login-form")
    print("Home page Title is ", driver.title)

    # Find the username field
    username = driver.find_element(By.XPATH, "//*[@id='username']")
    # Find the password field
    password = driver.find_element(By.XPATH, "//*[@id='password']")

    # Enter the given credentials
    # Enter username
    username.send_keys("admin")
    # Enter password
    password.send_keys("password")

    # Find the login button
    login = driver.find_element(By.XPATH, "/html/body/div[2]/div/div[2]/div[2]/div/div/button")
    login.click()

    # Print the login error message

    print("Login button clicked")

    # Perform testing and assertions
    
    # Close the browser
    driver.close()
